ci: bump ubuntu version for reusable_testing

Buristan opened the pull request:
This patch bumps the Ubuntu distro version up to 24.04 (noble) in the aforementioned workflow, since the Ubuntu 20.04 (focal) image support will be dropped soon in the GitHub actions.

See also: actions/runner-images#11101

Buristan committed:
ci: bump ubuntu version for reusable_testing

This patch bumps the Ubuntu distro version up to 24.04 (noble) in the
aforementioned workflow, since the Ubuntu 20.04 (focal) image support
will be dropped soon in the GitHub actions.

See also: actions/runner-images#11101

Buristan committed:
ci: actualize fast_testing workflow

This patch bumps the versions of Tarantool up to 2.11, 3.2 and 3.3 as
currently supported with built packages for the latest Ubuntu 24.04
(noble).

Also, it bumps the version of the `actions/checkout` actions to v4 and
the version of the `actions/setup-python` action to v5. It is needed for
fixing the following GitHub warnings:
| Node.js 16 actions are deprecated.
| Please update the following actions to use Node.js 20
|
| The following actions uses node12 which is deprecated and will be
| forced to run on node16

Also, it updates `setup-tarantool` action to v3. It is needed to make
Tarantool 3.x pipelines work, see [1] for details.

[1]: https://github.com/tarantool/setup-tarantool/releases/tag/v3.0.0

Totktonada commented on .github/workflows/fast_testing.yml:
It would be nice to also update `setup-tarantool` to `v3`.

Buristan replied:
Update to version 3:
```diff
diff --git a/.github/workflows/fast_testing.yml b/.github/workflows/fast_testing.yml
index cdc4960..30779fe 100644
--- a/.github/workflows/fast_testing.yml
+++ b/.github/workflows/fast_testing.yml
@@ -30,7 +30,7 @@ jobs:
           submodules: recursive
 
       - name: Setup tarantool ${{ matrix.tarantool }}
-        uses: tarantool/setup-tarantool@v1
+        uses: tarantool/setup-tarantool@v3
         with:
           tarantool-version: ${{ matrix.tarantool }}
 
```

Totktonada commented on .github/workflows/fast_testing.yml:
`tarantool-common` package seems broken (I've reproduced the CI failure locally in docker):

```
Errors were encountered while processing:
 /tmp/apt-dpkg-install-wzouzC/110-tarantool-common_2.11.6.g74ad6a412d-1_all.deb
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

Totktonada replied:
Filed https://github.com/tarantool/tarantool/issues/11234.

Totktonada replied:
I've re-deployed the fixed tarantool-common packages for 2.11.6 for Ubuntu 24.04 Noble.

Buristan committed:
ci: actualize osx testing workflow

This patch fixes a bunch of problems:
* Updates macOS runners, since macos-11 and macos-12 are
  unsupported, so there is no such runner to pick up the job.
* Updates the list of Tarantool versions.
* Bumps the `actions/cache` up to v4, since v2 is deprecated.
* Drops the version suffix from the key parameter for the cache, since
  it is not necessary after [1].
* Bumps the `actions/checkout` up to v4, since v2 is deprecated.
* Adds use of `actions/setup-python` of v5, to fixate the python 3.10
  for tests.
* Drops patching step, since it is not needed anymore and causes
  conflicts.

[1]: https://github.blog/changelog/2022-10-20-manage-caches-in-your-actions-workflows-from-web-interface/

Buristan commented:
I also add the commit to fix outdated OSX testing workflow.

Totktonada commented on .github/workflows/osx_testing.yml:
```diff
@@ -53,20 +42,16 @@ jobs:
         id: cache
         with:
           path: ${{ env.T_DESTDIR }}
-          # v2 is due to https://github.com/actions/cache/issues/2
-          # and because the cache keys without -v2 may contain
-          # debug tarantool builds. It is desirable to have all
-          # build either debug or release (RelWithDebInfo), but
-          # we unable to build all releases in debug (see below).
-          #
-          # v3 is to re-verify all Mac OS builds after fix for the
-          # gh-6076 problem (see below).
-          #
```
It is about the `key` version below, not `actions/cache@v<...>`.

It is not so necessary after https://github.blog/changelog/2022-10-20-manage-caches-in-your-actions-workflows-from-web-interface/ , so you can drop this versioning if you want.

Buristan replied:
Dropped it and updates the commit message:
```diff
diff --git a/.github/workflows/osx_testing.yml b/.github/workflows/osx_testing.yml
index 6144357..dc6c1e6 100644
--- a/.github/workflows/osx_testing.yml
+++ b/.github/workflows/osx_testing.yml
@@ -52,9 +52,7 @@ jobs:
         id: cache
         with:
           path: ${{ env.T_DESTDIR }}
-          # v4 added due to inability to clear the cache after v3 prefix.
-          # See https://github.com/github/docs/issues/14145
-          key: ${{ matrix.runs-on }}-${{ matrix.tarantool }}-v4
+          key: ${{ matrix.runs-on }}-${{ matrix.tarantool }}
         if: matrix.tarantool != 'brew' && matrix.tarantool != 'master'
 
       - name: Install tarantool build dependencies
```
